rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -2375,6 +2375,35 @@ public UserData getCurrentUserData() {
         return userData;
     }
 
+    @Override
+    public void releaseDanglingBusyNodes(List<NodeSet> busyNodesWithTask) {
+        Set<String> busyNodesUrlWithTask = new HashSet<>();
+
+        for (NodeSet nodeSet : busyNodesWithTask) {
+            busyNodesUrlWithTask.addAll(nodeSet.getAllNodesUrls());
+        }
+
+        List<RMNode> busyNodesToRelease = findDanglingBusyNodes(busyNodesUrlWithTask);
+
+        nodesCleaner.cleanAndRelease(busyNodesToRelease);
+    }
+
+    private List<RMNode> findDanglingBusyNodes(Set<String> busyNodesUrlWithTask) {
+        List<RMNode> busyNodesToRelease = new LinkedList<>();
+
+        for (Entry<String, RMNode> nodeEntry : allNodes.entrySet()) {
+            String nodeUrl = nodeEntry.getKey();
+            RMNode node = nodeEntry.getValue();
+
+            if (node.isBusy() && !busyNodesUrlWithTask.contains(nodeUrl)) {
+                logger.info("Dangling busy node found: " + node.getNodeName());
```

Comment:
I would prefer not to put the info log in the find method

Reply:
ok